# In-Class Practice Session I

__Mobimania is a smartphone retailers spread across the India. To perceive customer choice, 
they collected past months sales data along with the rating given by the customers. Find the 
answers to the following questions after reading Mobiles.csv separated by pipe(|):__
   
    Q1. Find the probability that a randomly picked customer rated his newly bought phone 4 out of 5.
    
    Q2. Find the probability that a phone has IPS display.
    
    Q3. Find the probability that a randomly bought phone has brand name Xiaomi.
    
    Q4. Find the probability that a phone has 23MP front and rear camera.
    
    Q5. Find the probability that a phone has 256GB ROM and 8GB RAM.
    
    Q6. If a randomly purchased phone has 64GB ROM, what will be the probability it has 4GB RAM?
    
    Q7. Given the phone has MicroSD Slot, find out the probability that it has 
        at least 3 star rating.

    Q8. If the phone has 3 star ratings, find the probability that it has at least 8 GB RAM.

    Q9. Given the phone has an IPS display, find the probability that it has a rear camera.

#### Data is available here:  https://www.dropbox.com/s/00d81prb1q7hxw2/Mobiles.csv

In [ ]:
import pandas as pd
!wget https://www.dropbox.com/s/00d81prb1q7hxw2/Mobiles.csv

--2021-03-28 16:45:19--  https://www.dropbox.com/s/00d81prb1q7hxw2/Mobiles.csv
Resolving www.dropbox.com (www.dropbox.com)... 162.125.5.18, 2620:100:601d:18::a27d:512
Connecting to www.dropbox.com (www.dropbox.com)|162.125.5.18|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: /s/raw/00d81prb1q7hxw2/Mobiles.csv [following]
--2021-03-28 16:45:19--  https://www.dropbox.com/s/raw/00d81prb1q7hxw2/Mobiles.csv
Reusing existing connection to www.dropbox.com:443.
HTTP request sent, awaiting response... 302 Found
Location: https://uc44a3125c1dfdb86d3ef44a4858.dl.dropboxusercontent.com/cd/0/inline/BLj-s59fcXR6C8fBAO3tVqvAyGesKAkFNDgJfnyxRc0GXpQa4fKl4cy8zEraT8Lhz-ZZ1PfhHmW456kzv4YN4keo1zgHmhSxSpEx7f46BaSrj89MZMFyi_Jn9OJ9pNXfzY43oVNwcR0_eHXj5oRIzito/file# [following]
--2021-03-28 16:45:20--  https://uc44a3125c1dfdb86d3ef44a4858.dl.dropboxusercontent.com/cd/0/inline/BLj-s59fcXR6C8fBAO3tVqvAyGesKAkFNDgJfnyxRc0GXpQa4fKl4cy8zEraT8Lhz-ZZ1PfhHmW456kzv4YN4keo1zgHm

In [ ]:
df = pd.read_csv('Mobiles.csv',sep='|')

In [ ]:
df.head()

PurchaseID       Brand  ... DisplayType Ratings
0  6dc8d963-8be1-4f16-9ce8-7241b671ecc6       Apple  ...      AMOLED       5
1  655dd155-3a4a-48af-832d-bd5fc7d24ba6   Microsoft  ...      AMOLED       3
2  fd5ee7a0-abcd-4fb0-88a9-581bc1d400f1  Blackberry  ...         IPS       3
3  90039014-2c01-4040-a835-3ab4501680ed  Blackberry  ...      AMOLED       4
4  a086de1b-9717-40a5-a4b9-1e9ec374670b   Panasonic  ...      AMOLED       5

[5 rows x 9 columns]

####   Q1. Find the probability that a randomly picked customer rated his newly bought phone 4 out of 5.
   

In [ ]:
s = df.shape[0]

In [ ]:
# We have 100 customers that makes:
S=1000

In [ ]:
df.DisplayType.value_counts()

AMOLED    511
IPS       489
Name: DisplayType, dtype: int64

In [ ]:
rdf = df.Ratings.value_counts()


In [ ]:
# Event: Customer rates the phone 4 out of 5.
e = rdf[4]
e

231

In [ ]:
print('Probabilty that a customer will rate the phone 4/5 is {}'.format(round(e/s,3)))

Probabilty that a customer will rate the phone 4/5 is 0.231


####    Q2. Find the probability that a phone has IPS display.
    
    

In [ ]:
# Event E: Phone has IPS Display
ddf = df.DisplayType.value_counts()
ddf

AMOLED    511
IPS       489
Name: DisplayType, dtype: int64

In [ ]:
E=ddf['IPS']
E

489

In [ ]:
print('Probability that a phone has IPS display {}'.format(round(E/s,3)))

Probability that a phone has IPS display 0.489


#### Q3. Find the probability that a randomly bought phone has brand name Xiaomi.
   

In [ ]:
# Event E: Phone has a brand name Xiaomi
bdf = data.Brand.value_counts()

In [ ]:
bE = bdf['Xiaomi']
bE

86

In [ ]:
print('Probability that a phone has brand name Xiaomi {}'.format(round(bE/s,3)))

Probability that a phone has brand name Xiaomi 0.086


####     Q4. Find the probability that a phone has 23MP front and rear camera.
    
  

In [ ]:
'''
Events:
Event E: Phone has 23MP front and rear camera.
Event A: Phone has 23MP front camera.
Event B: Phone has 23MP rear camera.
'''
fcdf = data.FrontCamera.value_counts()
fcdf

7 MP      96
No        84
2.1 MP    82
VGA       81
5.1 MP    78
8 MP      73
5 MP      72
2 MP      68
1.3 MP    67
13 MP     64
0.9 MP    64
23 MP     61
4 MP      56
1.2 MP    54
Name: FrontCamera, dtype: int64

In [ ]:
A=fcdf['23 MP']
A

61

In [ ]:
rcdf = data.RearCamera.value_counts()
rcdf

13 MP      75
20 MP      74
10 MP      74
3 MP       73
23 MP      73
0.3 MP     71
1.3 MP     70
20.7 MP    68
2 MP       67
16 MP      66
5 MP       64
8 MP       63
12 MP      61
3.15 MP    58
No         42
21 MP       1
Name: RearCamera, dtype: int64

In [ ]:
B=rcdf['23 MP']
B

73

In [ ]:
e = df[(df.FrontCamera=='23 MP') & (df.RearCamera=='23 MP')]
e

PurchaseID      Brand  ... DisplayType Ratings
24   6d68342f-02e1-4578-9ceb-a427a0f71d86   Micromax  ...         IPS       4
51   d15ca9fb-f87e-4a1a-813b-03c85bdda867  Microsoft  ...         IPS       1
64   8a3d164e-d315-4031-8918-fc5f7ac117cc         LG  ...      AMOLED       4
178  a180e608-df52-49ec-98a2-95264faea870      Vertu  ...         IPS       2
761  461ea69c-736c-49f7-b2c5-581ddf015231   Micromax  ...      AMOLED       4

[5 rows x 9 columns]

In [ ]:
E=A+B
E

134

In [ ]:
print('Probability that a phone has 23 MP front and rear camera {}'.format(round(E/s,3)))

Probability that a phone has 23 MP front and rear camera 0.134


####  Q5. Find the probability that a phone has 256GB ROM and 8GB RAM.
    
    

In [ ]:
'''
Events:
Event E: Phone has 256GB ROM and 8GB RAM.
Event A: Phone has 256GB ROM.
Event B: Phone has 8GB RAM.
'''
df.ROM.value_counts()

256 GB    190
128 GB    165
16 GB     164
8 GB      164
32 GB     159
64 GB     158
Name: ROM, dtype: int64

In [ ]:
df.RAM.value_counts()

16 GB     111
3 GB      103
64 GB      99
128 GB     97
4 GB       93
1 GB       90
8 GB       90
2 GB       83
32 GB      83
256 GB     80
512 MB     71
Name: RAM, dtype: int64

In [ ]:
dfrr = df[(df.RAM=='4 GB')&(df.ROM=='64 GB')]
len(dfrr)/158

0.06962025316455696

In [ ]:
len(data[(data.RAM=='8 GB')&(data.ROM=='256 GB')])

20

In [ ]:
E=20

In [ ]:
print('Probability that a phone has 256GB ROM and 8GB RAM {}'.format(round(E/S,3)))

Probability that a phone has 256GB ROM and 8GB RAM 0.02


#### Q6. If a randomly purchased phone has 64GB ROM, what will be the probability it has 4GB RAM?
    
   

In [ ]:
'''
Events:
Event E: If phone has 64GB ROM, then it also has 4GB RAM
Event A: Phone has 64GB ROM
Event B: Phone has 4GB RAM.
'''
data.ROM.value_counts()

256 GB    190
128 GB    165
16 GB     164
8 GB      164
32 GB     159
64 GB     158
Name: ROM, dtype: int64

In [ ]:
A=158

In [ ]:
probability_A=round(A/S,3)

In [ ]:
# Let's see joint event when phone has both 64GB ROM and 4GB RAM
pd.crosstab(data.ROM,data.RAM, margins=True)

RAM     1 GB  128 GB  16 GB  2 GB  256 GB  ...  4 GB  512 MB  64 GB  8 GB   All
ROM                                        ...                                 
128 GB    11      11     20    14      19  ...    14      16     18    12   165
16 GB     15      16     17    13      14  ...    21      12     14    11   164
256 GB    20      19     23    18      15  ...    13      13     20    20   190
32 GB     14      18     18    10      13  ...    17       9     22    12   159
64 GB     15      18     18    12       8  ...    11      10     11    18   158
8 GB      15      15     15    16      11  ...    17      11     14    17   164
All       90      97    111    83      80  ...    93      71     99    90  1000

[7 rows x 12 columns]

In [ ]:
A_and_B=11

In [ ]:
probability_A_and_B=round(A_and_B/S,3)

In [ ]:
print('Joint probability that a phone has 64GB ROM and 4GB RAM: ',probability_A_and_B)

Joint probability that a phone has 64GB ROM and 4GB RAM:  0.011


In [ ]:
print('Probability that a phone has 64GB ROM, then it also has 4GB RAM {}'.format(
    round(probability_A_and_B/probability_A,3)))

Probability that a phone has 64GB ROM, then it also has 4GB RAM 0.07


####  Q7. Given the phone has MicroSD Slot, find out the probability that it has at least 3 star rating.

    

In [ ]:
'''
Events:
Event E: If phone has MicroSD Slot, then it has atleast 3 star rating.
Event A: Phone has MicroSD Slot
Event B: Phone has 3 star rating.
'''
data.MicroSD.value_counts()

True     506
False    494
Name: MicroSD, dtype: int64

In [ ]:
A=506

In [ ]:
probability_A=round(A/S,3)

In [ ]:
print('Probability that a phone has MicroSD Slot {}'.format(probability_A))

Probability that a phone has MicroSD Slot 0.506


In [ ]:
# Let's see joint event when phone has MicroSD SLot and 3 star rating
pd.crosstab(data.MicroSD,data.Ratings, margins=True)

Ratings    1    2    3    4    5   All
MicroSD                               
False     88   90  100  120   96   494
True     110   84   95  111  106   506
All      198  174  195  231  202  1000

In [ ]:
A_and_B=95

In [ ]:
probability_A_and_B=round(A_and_B/s,3)
probability_A_and_B

0.095

In [ ]:
print('Probability that phone has MicroSD Slot and 3 star rating {}'.format(probability_A_and_B))

Probability that phone has MicroSD Slot and 3 star rating 0.031


In [ ]:
print('Probability that if phone has MicroSD Slot, then it has atleast 3 star rating {}'.format(
        round(probability_A_and_B/probability_A,3)))

Probability that if phone has MicroSD Slot, then it has atleast 3 star rating 0.188


#### Q8. If the phone has 3 star ratings, find the probability that it has at least 8 GB RAM.

    

In [ ]:
'''
Events:
Event E: If phone has 3 star rating, then it has at leat 8 GB ROM.
Event A: Phone has 3 star rating.
Event B: Phone has 8 GB ROM.
'''
data.Ratings.value_counts()

4    231
5    202
1    198
3    195
2    174
Name: Ratings, dtype: int64

In [ ]:
A=195

In [ ]:
probability_A=round(A/S,3)

In [ ]:
print('Probability that a phone has 3 star rating {}'.format(probability_A))

Probability that a phone has 3 star rating 0.195


In [ ]:
# Let's see joint event when phone has 3 star rating and 8 GB ROM
pd.crosstab(data.Ratings,data.ROM, margins=True)

ROM      128 GB  16 GB  256 GB  32 GB  64 GB  8 GB   All
Ratings                                                 
1            32     35      30     41     25    35   198
2            29     23      36     29     31    26   174
3            27     36      47     29     28    28   195
4            43     38      41     34     34    41   231
5            34     32      36     26     40    34   202
All         165    164     190    159    158   164  1000

In [ ]:
A_and_B=28

In [ ]:
probability_A_and_B=round(A_and_B/S,3)

In [ ]:
print('Probability that a phone has 3 star rating and 8 GB ROM {}'.format(probability_A_and_B))

Probability that a phone has 3 star rating and 8 GB ROM 0.028


In [ ]:
print('Probability that if phone has 3 star rating, then it has at leat 8 GB ROM {}'.format(
        round(probability_A_and_B/probability_A,3)))

Probability that if phone has 3 star rating, then it has at leat 8 GB ROM 0.144


#### Q9. Given the phone has an IPS display, find the probability that it has a 8 MP rear camera.

    

In [ ]:
'''
Events:
Event E: If phone has IPS display, then it has a 8 MP camera
Event A: Phone has IPS display
Event B: Phone has 8 MP camera
'''
data.DisplayType.value_counts()

AMOLED    511
IPS       489
Name: DisplayType, dtype: int64

In [ ]:
A=489

In [ ]:
probability_A=round(A/S,3)

In [ ]:
print(' Probability that a phone has  IPS display {}'.format(probability_A))

 Probability that a phone has  IPS display 0.489


In [ ]:
pd.crosstab(data.DisplayType,data.RearCamera,margins=True)

RearCamera   0.3 MP  1.3 MP  10 MP  12 MP  13 MP  ...  3.15 MP  5 MP  8 MP  No   All
DisplayType                                       ...                               
AMOLED           42      36     45     33     38  ...       21    35    32  21   511
IPS              29      34     29     28     37  ...       37    29    31  21   489
All              71      70     74     61     75  ...       58    64    63  42  1000

[3 rows x 17 columns]

In [ ]:
A_and_B=31

In [ ]:
probability_A_and_B=round(A_and_B/S,3)

In [ ]:
print('Probability phone has IPS display and 8 MP Rear Camera',probability_A_and_B)

Probability phone has IPS display and 8 MP Rear Camera 0.031


In [ ]:
print('Probability that if phone has IPS display, then it has a 8 MP camera ',round(probability_A_and_B/probability_A,3))

Probability that if phone has IPS display, then it has a 8 MP camera  0.063


# In-Class Practice Session II

Q1. Men heights have a mean of 63.6 inches and a standard deviation of 2.5 inches.  
	 Find the z-score corresponding to a man with a height of 78 inches.

Q2. The local electricity authority conducted a study on electrical bills. It was found that average bill amount
	 was 527 with a standard deviation of 112. What is the probability that a citizen’s bill will be more
	 than 500? 

Q3. The average IQ of a human being is 100 and standard deviation of 16. A certain individual has an IQ of 160.
 
- What is the difference between Individual’s IQ and the mean? 
- How many standard deviations is that? 
- Convert Individual’s IQ score to a z score. 
- If we consider, usual IQ scores to be those that  z-scores between -2 and 2, 
       then is the Individual’s IQ usual or unusual?  


#### Q1. Men heights have a mean of 63.6 inches and a standard deviation of 2.5 inches.
#### Find the z-score corresponding to a man with a height of 78 inches.

![image.png](attachment:image.png)

In [ ]:
# In this case we just have calculate the z score:
X = 78
mean = 63.6
std = 2.5

In [ ]:

Z = (X-mean)/std

In [ ]:
Z

5.76

#### Q2. The local electricity authority conducted a study on electrical bills. It was found that average bill amount was 527 with a standard deviation of 112. What is the probability that a citizen’s bill will be more than 500?

In [ ]:
mean = 527
std = 112
X = 500

In [ ]:
# Let's calculate z score

Z = (X-mean)/std

In [ ]:
print(round(Z,3))

-0.241


For next step, we can use the Z-score table from here: https://www.dummies.com/education/math/statistics/how-to-use-the-z-table/

we get __P(Z<-.241)=0.4052__

to calculate __P(Z>-0.241) = 1 - P(Z<-0.241)__

In [ ]:
print('Probability that customer\'s bill will be more than 500: ',1-0.4052)

Probability that customer's bill will be more than 500:  0.5948


Q3. The average IQ of a human being is 100 and standard deviation of 16. A certain individual has an IQ of 160.
<ol> 
<li> What is the difference between Individual’s IQ and the mean? 
<li> How many standard deviations is that? 
<li> Convert Individual’s IQ score to a z score. 
<li> If we consider, usual IQ scores to be those that  z-scores between -2 and 2, 
       then is the Individual’s IQ usual or unusual? 

#### 1. What is the difference between Individual’s IQ and the mean?

In [ ]:
ind_IQ=160
mean=100
print('difference between Individual’s IQ and the mean',ind_IQ-mean)

difference between Individual’s IQ and the mean 60


#### 2. How many standard deviations is that?

In [ ]:
std=16

In [ ]:
print('{} number of standard deviations'.format(60/16))

3.75 number of standard deviations


#### 3. Convert Individual’s IQ score to a z score.

In [ ]:
Z=(160-100)/16

In [ ]:
print('Z-score',Z)

Z-score 3.75


#### 4. If we consider, usual IQ scores to be those that z-scores between -2 and 2, then is the Individual’s IQ usual or unusual?

since Z-score is more than the range given, we can consider it as __Unusual__.